osv2/index_create.go

```diff
@@ -0,0 +1,161 @@
+package osv2
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"time"
+
+	"github.com/opensearch-project/opensearch-go/v2"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchapi"
+
+	"github.com/CrowdStrike/opensearchtools"
+)
+
+// CreateIndexRequest is a domain model union type for all the fields of CreateIndexRequests for all
+// supported OpenSearch versions.
+// Currently supported versions are:
+//   - OpenSearch 2
+//
+// An empty CreateIndexRequest will fail to execute. At least index is required to be added which matches to
+// the existing template index pattern. Otherwise, the DocBody has to be provided with the detailed index information
+// as provided in the documentation: [CreateIndex] https://opensearch.org/docs/latest/api-reference/index-apis/create-index/
+type CreateIndexRequest struct {
+	Index               string
+	DocBody             io.Reader
+	MasterTimeout       time.Duration
+	Timeout             time.Duration
+	WaitForActiveShards string // todo: update this with enum or not since we have numbers and all
+}
+
+// FromDomainCreateIndexRequest creates a new [BulkRequest] from the given [opensearchtools.CreateIndexRequest]
+func FromDomainCreateIndexRequest(req *opensearchtools.CreateIndexRequest) (CreateIndexRequest, opensearchtools.ValidationResults) {
+	// As more versions are implemented, these [opensearchtools.ValidationResults] may be used to contain issues
+	// converting from the domain model to the V2 model.
+	var vrs opensearchtools.ValidationResults
+
+	return CreateIndexRequest{
+		Index:               req.Index,
+		DocBody:             req.DocBody,
+		MasterTimeout:       req.MasterTimeout,
+		Timeout:             req.Timeout,
+		WaitForActiveShards: req.WaitForActiveShards,
+	}, vrs
+}
+
+// Validate validates the given CreateIndexRequest
+func (c *CreateIndexRequest) Validate() opensearchtools.ValidationResults {
+	var validationResults opensearchtools.ValidationResults
+
+	if c.Index == "" {
+		validationResults.Add(opensearchtools.NewValidationResult("Index not set at the CreateIndexRequest", true))
+	}
+
+	return validationResults
+}
+
+// NewCreateIndexRequest instantiates an CreateIndexRequest with default values
+func NewCreateIndexRequest() *CreateIndexRequest {
+	return &CreateIndexRequest{
+		MasterTimeout:       30 * time.Second,
+		Timeout:             30 * time.Second,
+		WaitForActiveShards: "1",
+	}
+}
+
+// WithIndex adds the index for CreateIndexRequest
```

Comment:
```suggestion
// WithIndex sets the index for CreateIndexRequest
```
Use "sets".  Using "adds" implies that the new value will be in addition to any existing value.